test/pspm_check_python_test.m

```diff
@@ -0,0 +1,62 @@
+classdef pspm_check_python_test < matlab.unittest.TestCase
+    % unittest class for the pspm_check_python function
+    % PsPM TestEnvironment
+    % ● Authorship
+    % Abdul Wahab Madni 2024(Uni Bonn) 
+
+    methods (TestMethodSetup)
+        function addFunctionPath(testCase)
+            % Add the path to the source directory
+            src_path = fullfile(pwd, '..', 'src');
+            addpath(src_path);
+        end
+    end
+
+    methods (TestMethodTeardown)
+        function removeFunctionPath(testCase)
+            % Remove the path to the source directory
+            src_path = fullfile(pwd, '..', 'src');
+            rmpath(src_path);
+        end
+    end
+
+    methods (Test)
+        function test_current_python_environment(this)
+            % Test case to report the current Python environment if it exists
+            % Setup: ensure a Python environment is already configured
+            original_env = pyenv;
+            if original_env.Version ~= ""  % Check if there's an existing Python environment
+                sts = pspm_check_python();
+                this.verifyEqual(sts, 1, 'Test failed: Expected sts = 1 when reporting the current Python environment');
+            else
+                this.assumeFail('No Python environment set up for testing reporting current environment.');
+            end
+            pyenv('Version', original_env.Version);  % Reset to original Python environment after the test
+        end
+
+        function test_set_new_python_environment(this)
+            % Test setting a new Python environment
+            % Note: Insert a valid path for the Python executable in your system
+            valid_python_path = 'C:\Users\TRA 3 PC0032\AppData\Local\Programs\Python\Python310\python.exe';  % Example path, change to a valid one for the test environment
```

Comment:
This can be simplified to `~/AppData/...`.
PsPM supports Win/Mac/linux. Previous build has considered Mac and Linux as much it can. Not sure if Mac and linux are considered in this occasion.

Reply:
Actually the path is given by the user in the original function and the function works on it accordingly and sets or checks the environment. But when making test, I had to provided my path to test the function that's why I have given a windows path. You have to change the path accoring to your system.